server/src/main/java/io/druid/client/BrokerServerView.java

```diff
@@ -188,6 +191,11 @@ private QueryableDruidServer removeServer(DruidServer server)
 
   private void serverAddedSegment(final DruidServerMetadata server, final DataSegment segment)
   {
+    if (segmentWatcherConfig.getWatchedTiers() != null && !segmentWatcherConfig.getWatchedTiers()
+                                                                               .contains(server.getTier())) {
```

Comment:
hmmm, dint think it would ever be null. by default all things are in default tier.You are suggesting this for defensive coding or is there really a scenario where getTier() returns null ?